framework/src/executor/mod.rs

```diff
@@ -234,9 +234,25 @@ impl<S: 'static + Storage, DB: 'static + TrieDB, Mapping: 'static + ServiceMappi
             .service_mapping
             .get_service(context.get_service_name(), sdk)?;
 
+        let mut tx_hook_services = vec![];
+        for name in self.service_mapping.list_service_name().into_iter() {
+            let sdk = self.get_sdk(&name)?;
+            let tx_hook_service = self.service_mapping.get_service(name.as_str(), sdk)?;
+            tx_hook_services.push(tx_hook_service);
+        }
+
         let result = match exec_type {
             ExecType::Read => service.read_(context),
-            ExecType::Write => service.write_(context),
+            ExecType::Write => {
+                for tx_hook_service in tx_hook_services.iter_mut() {
```

Comment:
I doubt this is not the right place for executor to exec tx_hook logic. Tx_hook( before or after) logic should be executed only one time for a tx lifetime. But if logic code wrote in here,  it will be executed every time when a service call another service , because `fn sdk.write` will trigger  it.

Reply:
Insert the `tx_hook` here. https://github.com/nervosnetwork/muta/blob/fda4204f52b4214078b1ef5894b950b1937ea778/framework/src/executor/mod.rs#L290